src/qibolab/instruments/qm/config.py

```diff
@@ -209,39 +216,42 @@ def register_flux_element(self, qubit, intermediate_frequency=0):
                 will send to this qubit. This frequency will be mixed with the
                 LO connected to the same channel.
         """
-        if f"flux{qubit.name}" not in self.elements:
-            self.elements[f"flux{qubit.name}"] = {
-                "singleInput": {
-                    "port": qubit.flux.port.pair,
-                },
-                "intermediate_frequency": intermediate_frequency,
-                "operations": {},
-            }
-        else:
-            self.elements[f"flux{qubit.name}"][
-                "intermediate_frequency"
-            ] = intermediate_frequency
+        element = f"flux{qubit.name}"
+        if element in self.elements:
+            return self._new_frequency_element(qubit, intermediate_frequency, "flux")
+
+        self.elements[element] = {
+            "singleInput": {
+                "port": qubit.flux.port.pair,
+            },
+            "intermediate_frequency": intermediate_frequency,
+            "operations": {},
+        }
+        return element
 
     def register_element(self, qubit, pulse, time_of_flight=0, smearing=0):
         if pulse.type is PulseType.DRIVE:
             # register drive element
             if_frequency = pulse.frequency - math.floor(qubit.drive.lo_frequency)
-            self.register_drive_element(qubit, if_frequency)
+            element = self.register_drive_element(qubit, if_frequency)
             # register flux element (if available)
             if qubit.flux:
                 self.register_flux_element(qubit)
         elif pulse.type is PulseType.READOUT:
             # register readout element (if it does not already exist)
             if_frequency = pulse.frequency - math.floor(qubit.readout.lo_frequency)
-            self.register_readout_element(qubit, if_frequency, time_of_flight, smearing)
+            element = self.register_readout_element(
+                qubit, if_frequency, time_of_flight, smearing
+            )
             # register flux element (if available)
             if qubit.flux:
                 self.register_flux_element(qubit)
         else:
             # register flux element
-            self.register_flux_element(qubit, pulse.frequency)
+            element = self.register_flux_element(qubit, pulse.frequency)
+        return element
 
-    def register_pulse(self, qubit, qmpulse):
+    def register_pulse(self, qubit, qmpulse, element=None):
```

Comment:
I don't believe you're using this `element` argument anywhere (instead, you're reading `qmpulse.element`).

Reply:
Fixed in #908.